File: app/src/main/java/com/druger/refresher/adapter/TabAdapter.java
package com.druger.refresher.adapter;

import android.app.Fragment;
import android.app.FragmentManager;
import android.support.v13.app.FragmentStatePagerAdapter;

import com.druger.refresher.fragments.CurrentTaskFragment;
import com.druger.refresher.fragments.DoneTaskFragment;

/**
 * Created by druger on 12.09.2015.
 */
public class TabAdapter extends FragmentStatePagerAdapter {
    private int numberOfTabs;

    public static final int CURRENT_TASK_FRAGMENT_POSITION = 0;
    public static final int DONE_TASK_FRAGMENT_POSITION = 1;

    private CurrentTaskFragment currentTaskFragment;
    private DoneTaskFragment doneTaskFragment;

    public TabAdapter(FragmentManager fm, int numberOfTabs) {
        super(fm);
        this.numberOfTabs = numberOfTabs;
        currentTaskFragment = new CurrentTaskFragment();
        doneTaskFragment = new DoneTaskFragment();
    }

    @Override
    public Fragment getItem(int i) {
        switch (i){
            case 0:
                return currentTaskFragment;
            case 1:
                return doneTaskFragment;
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return numberOfTabs;
    }
}

File: app/src/main/java/com/druger/refresher/adapters/TabAdapter.kt
package com.druger.refresher.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.druger.refresher.ui.fragments.CurrentTaskFragmentNew
import com.druger.refresher.ui.fragments.DoneTaskFragmentNew

/**
* Created by druger on 12.09.2015.
*/
class TabAdapter(
    fm: FragmentManager, private var numberOfTabs: Int
) : FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    companion object {
        const val CURRENT_TASK_FRAGMENT_POSITION = 0
        const val DONE_TASK_FRAGMENT_POSITION = 1
    }

    private var currentTaskFragment = CurrentTaskFragmentNew()
    private var doneTaskFragment = DoneTaskFragmentNew()


    override fun getItem(position: Int): Fragment {
        return if (position == 0) currentTaskFragment else doneTaskFragment
    }

    override fun getCount(): Int {
        return numberOfTabs
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return when(position) {
            0 -> "Current"
            1 -> "Done"
            else -> super.getPageTitle(position)
        }
    }
}
